lumigator/python/mzai/backend/backend/api/routes/jobs.py

```diff
@@ -103,20 +105,57 @@ def list_jobs(
     )
 
 
-@router.get("/{job_id}")
-def get_job(service: JobServiceDep, job_id: UUID) -> Job:
+@router.get("/{job_spec}")
```

Comment:
what will a spec be in this case? we could consider a more descriptive variable name here

Reply:
i.e. are we changing this from a UUID?

Reply:
We are. It can be a uuid (returning a single job) or a job type name (`inference` or `evaluate`, returning a list of jobs). It follows the approach in the create_job service for consistency. I'd rather use a param (actually inheritance via anyOf plus a discriminator).

Reply:
I'd prefer to have one route return one type of object for clarity and consistency. If we have a spec that returns multiple objects, we should return a dictionary of elements.

Reply:
We have a couple of constraints: make it backwards compatible, and follow a similar interface to the existing calls.
Regarding the dictionary, the `GET /jobs` call already returns a list, so I'd see it natural to return a list as well.
Regarding the routes, I can use a more specific route like `GET /jobs/per_type/{job_type}` to avoid overloading the `/jobs/{job_id}` route. Wdyt?

Reply:
We can actually use `/jobs/{uuid}` and `/jobs/inference/` / `/jobs/evaluate/` with trailing slashes :-/ It's a bit messy too but I kinda like it better.

Reply:
I agreed with @aittalam on using `/jobs/{job_type}/` to keep it consistent with the existing POST methods. Let me know if this would be ok for you.